app/http.php

```diff
@@ -0,0 +1,1039 @@
+        }
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_OK)
+            ->send();
+    });
+
+App::post('/v1/execution')
+    ->desc('Create an execution')
+    // Execution-related
+    ->param('runtimeId', '', new Text(64), 'The runtimeID to execute.')
+    ->param('payload', '', new Text(8192), 'Data to be forwarded to the function, this is user specified.', true)
+    ->param('variables', [], new Assoc(), 'Environment variables required for the build and execution.', true)
+    ->param('timeout', 15, new Range(1, (int) App::getEnv('OPEN_RUNTIMES_EXECUTOR_MAX_TIMEOUT', "900")), 'Function maximum execution time in seconds.', true)
+    // Runtime-related
+    ->param('image', '', new Text(128), 'Base image name of the runtime.', true)
+    ->param('source', '', new Text(0), 'Path to source files.', true)
+    ->param('entrypoint', '', new Text(256), 'Entrypoint of the code file.', true)
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(
+        function (string $runtimeId, string $payload, array $variables, int $timeout, string $image, string $source, string $entrypoint, Table $activeRuntimes, Response $response) {
+            $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
+            $runtimeId = System::getHostname() . '-' . $runtimeId; // Used in Docker (name)
+
+            $variables = \array_merge($variables, [
+                'INERNAL_EXECUTOR_HOSTNAME' => System::getHostname()
+            ]);
+
+            // Prepare runtime
+            if (!$activeRuntimes->exists($activeRuntimeId)) {
+                if (empty($image) || empty($source) || empty($entrypoint)) {
+                    throw new Exception('Runtime not found. Please start it first or provide runtime-related parameters.', 401);
+                }
+
+                // Prepare request to executor
+                $sendCreateRuntimeRequest = function () use ($activeRuntimeId, $image, $source, $entrypoint, $variables) {
+                    $statusCode = 0;
+                    $errNo = -1;
+                    $executorResponse = '';
+
+                    $ch = \curl_init();
+
+                    $body = \json_encode([
+                        'runtimeId' => $activeRuntimeId,
+                        'image' => $image,
+                        'source' => $source,
+                        'entrypoint' => $entrypoint,
+                        'variables' => $variables
+                    ]);
+
+                    \curl_setopt($ch, CURLOPT_URL, "http://localhost/v1/runtimes");
+                    \curl_setopt($ch, CURLOPT_POST, true);
+                    \curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+                    \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+                    \curl_setopt($ch, CURLOPT_TIMEOUT, ((int) App::getEnv('OPEN_RUNTIMES_EXECUTOR_BUILD_TIMEOUT', '900')) + 2); // max 2 seconds expected network latency
```

Comment:
```suggestion
                    \curl_setopt($ch, CURLOPT_TIMEOUT, ((int) App::getEnv('OPEN_RUNTIMES_BUILD_TIMEOUT', '900')) + 2); // max 2 seconds expected network latency
```

`EXECUTOR` is redundant here. `OPEN_RUNTIMES_BUILD_TIMEOUT` is clear in itself

Reply:
I prefix all variables for proxy with `OPEN_RUNTIMES_PROXY` and all in executor with `OPEN_RUNIMES_EXECUTOR`.

Are you sure we want to remove `executor` from here? It indicates this is value used by executor, not a proxy.

If we want to do this change I am sure there are many others that would make sense without `executor` too, like `OPEN_RUNTIMES_EXECUTOR_INACTIVE_TRESHOLD`.

Reply:
@christyjacob4 I agree with @Meldiron, we can keep it for now.

Reply:
We will replace `OPEN_RUNTIMES_`with `OPR_` to make it shorter.

Reply:
We shoudl remove max/limit variables all-together

Reply:
Renamed vars. Removed all limits, replaced them with endpoint params